checker/src/main/java/org/checkerframework/checker/optional/OptionalVisitor.java

```diff
@@ -328,9 +360,121 @@ public void handleConditionalStatementIsPresentGet(IfTree tree) {
   public Void visitMethodInvocation(MethodInvocationTree tree, Void p) {
     handleCreationElimination(tree);
     handleNestedOptionalCreation(tree);
+    updateCalleesToCallers(tree);
     return super.visitMethodInvocation(tree, p);
   }
 
+  /**
+   * Updates {@link calleesToCallers} given a method invocation.
+   *
+   * <p>Check whether the method is in the set of methods that must be checked by the Non-Empty
+   * checker whenever a method invocation is encountered. If the method is in the set, its caller
+   * should also be checked by the Non-Empty Checker.
+   *
+   * <p>This ensures that the <i>clients</i> of any methods that must be checked by the Non-Empty
+   * Checker (i.e., methods that have preconditions related to the Non-Empty type system) are
+   * included in the set of methods to check.
+   *
+   * @param tree a method invocation tree
+   */
+  private void updateCalleesToCallers(MethodInvocationTree tree) {
+    String callee = tree.getMethodSelect().toString();
+    MethodTree caller = TreePathUtil.enclosingMethod(this.getCurrentPath());
+    if (caller != null) {
+      // Using the names of methods (as opposed to their fully-qualified name or signature) is a
+      // safe (but imprecise) over-approximation of all the methods that must be verified with the
+      // Non-Empty Checker. Overloads of methods will be detected.
+      Set<String> namesOfMethodsForNonEmptyChecker =
+          methodsToVerifyWithNonEmptyChecker.stream()
+              .map(MethodTree::getName)
+              .map(Name::toString)
+              .collect(Collectors.toSet());
+      if (namesOfMethodsForNonEmptyChecker.contains(callee)) {
```

Comment:
You can integrate this into the stream methods in order to avoid creating a set and avoid full iteration if found.